### PR TITLE
Introduce ENABLE_OMP_TASKLOOP for enable/disable omp taskloop

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
NVHPC 21.3 (released after 3.11 release) stops the compilation when it hits "omp taskloop" with an error saying this feature is not supported.
To make the compiler still being able to build and test the code. Introduce this flag.

## What type(s) of changes does this code introduce?
- Build related changes

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

File: CMake/PGICompilers.cmake [hunk 1]
```diff
@@ -1,3 +1,7 @@
+if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 21.3 AND ENABLE_OMP_TASKLOOP)
```
Comment (prckent) [inline CMake/PGICompilers.cmake:1]: Why CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 21.3 ? Does 21.2 not abort or somehow implement the feature?

A more cmake-like and robust solution would be to test for the feature availability and not test for any specific compiler version. i.e. Test compile a have_omp_taskloop.cpp code, set a configure variable. Then the code will naturally start working when 22.0 comes along, or whichever version supports it. Keep ENABLE_OMP_TASKLOOP for manual testing and workarounds.

**Review by prckent — CHANGES_REQUESTED** [review]
See comment above.

I prefer to test for feature availability as well as have the flag.

File: CMake/PGICompilers.cmake (hunk 1 above)
Comment (ye-luo) [reply CMake/PGICompilers.cmake:1]: Good point on testing the feature. Will change.

Comment (ye-luo) [reply CMake/PGICompilers.cmake]: Added a feature test. 21.2 doesn't abort. I don't think the feature is implemented but probably no-op in 21.2. Once there is a bit of tasking (probably due to target nowait) in the runtime, no-op is wrong for taskloop. My workaround is setting taskgroup.

**Comment by ye-luo** [comment]
@prckent is this good now?

**Review by prckent — APPROVED** [review]

**Pull request merged**